tests/test_xice_calendar.py

```diff
@@ -0,0 +1,69 @@
+from unittest import TestCase
+import pandas as pd
+
+from .test_trading_calendar import ExchangeCalendarTestBase
+from trading_calendars.exchange_calendar_xice import XICEExchangeCalendar
+
+
+class XICECalendarTestCase(ExchangeCalendarTestBase, TestCase):
+
+    answer_key_filename = 'xice'
+    calendar_class = XICEExchangeCalendar
+
+    # The XICE is open from 9:30 am to 3:30 pm.
+    MAX_SESSION_HOURS = 6.0
+
+    HAVE_EARLY_CLOSES = False
+
+    # Note: Iceland does not observe DST
+    DAYLIGHT_SAVINGS_DATES = ['2018-04-05', '2018-11-01']
+
+    def test_all_holidays(self):
+        all_sessions = self.calendar.all_sessions
+
+        expected_holidays = [
+            pd.Timestamp('2018-01-01', tz='UTC'),  # New Year's Day
+            pd.Timestamp('2018-03-29', tz='UTC'),  # Maundy Thursday
+            pd.Timestamp('2018-03-30', tz='UTC'),  # Good Friday
+            pd.Timestamp('2018-04-02', tz='UTC'),  # Easter Monday
+            pd.Timestamp('2018-04-19', tz='UTC'),  # First Day of Summer
+            pd.Timestamp('2018-05-01', tz='UTC'),  # Labour Day
+            pd.Timestamp('2018-05-10', tz='UTC'),  # Ascension Day
+            pd.Timestamp('2019-06-17', tz='UTC'),  # National Day
```

Comment:
Did you mean 2018 here?

Reply:
also I think you are missing Whit Monday (2018-05-21).  Can you add that?

Reply:
National Day is tested for 2019, as it fell on a Sunday on 2018. I changed it to 2018 to `test_holidays_fall_on_weekend` so it's clearer.

I forgot Whit Monday - good catch (due to basing the test on XSTO). It's a holiday in the CSV test resource though.

Reply:
got it re: National Day, makes sense, thanks.